psi-sample/enclave/src/main/java/com/r3/conclave/sample/enclave/PSIEnclave.java

```diff
@@ -0,0 +1,117 @@
+package com.r3.conclave.sample.enclave;
+
+import com.esotericsoftware.kryo.Kryo;
+import com.esotericsoftware.kryo.io.Input;
+import com.r3.conclave.enclave.Enclave;
+import com.r3.conclave.mail.EnclaveMail;
+import com.r3.conclave.sample.common.AdDetails;
+import com.r3.conclave.sample.common.InputData;
+import com.r3.conclave.sample.common.InputDataSerializer;
+import com.r3.conclave.sample.common.UserDetails;
+
+import java.io.ByteArrayInputStream;
+import java.security.PublicKey;
+import java.util.*;
+import java.util.stream.Collectors;
+
+/**
+ * This class is the Enclave class. To create an enclave in Conclave extend your class by Enclave interface.
+ * This class will contain the our business logic to calculate the ad conversion rate by using the list of credit card numbers given by the merchant
+ * and the service provider. Since the lists are shared with the enclave and not with anyone else, complete privacy of data is guaranteed.
+ */
+public class PSIEnclave extends Enclave {
```

Comment:
Do we need to add any tests?

Reply:
Added MockTest to unit test enclave ad conversion logic.